cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java

```
@@ -2,38 +2,29 @@
 
 public class MortgageInstallmentCalculator {
 
-    /**
-     *
-     * @param p principal amount
-     * @param t term of mortgage in years
-     * @param r rate of interest
-     * @return monthly payment amount
-     */
-    public static double calculateMonthlyPayment(
-            int p, int t, double r) {
+    private static final String MESSAGE_BAD_INPUT_PARAMETERS = "Negative values are not allowed";
+    private static final int AMOUNT_MONTHS_IN_YEAR = 12;
 
-        //cannot have negative loanAmount, term duration and rate of interest
-        if (p < 0 || t <= 0 || r < 0) {
-            throw new InvalidInputException("Negative values are not allowed");
+    public static double calculateMonthlyPayment(int startAmount, int termInYears, double percent) {
+        if (haveValidParameters(startAmount, termInYears, percent)) {
+            throw new InvalidInputException(MESSAGE_BAD_INPUT_PARAMETERS);
```

you can throw this exception inside function `haveValidParameters`

In this case better to rename this function